tensorflow/python/ops/image_grad_test.py

```diff
@@ -456,6 +458,38 @@ def testGradRandomBoxes(self):
 
               self.assertLess(err, 2e-3)
 
+@test_util.run_all_in_graph_and_eager_modes
+class RGBToHSVOpTest(test.TestCase):
+
+  TYPES = [np.float32, np.float64]
+
+  def testShapeIsCorrectAfterOp(self):
+    in_shape = [2, 20, 30, 3]
+    out_shape = [2, 20, 30, 3]
+
+    for nptype in self.TYPES:
+      x = np.random.randint(0, high=255, size=[2, 20, 30, 3]).astype(nptype)
+      with self.cached_session(use_gpu=True):
+        rgb_input_tensor = constant_op.constant(x, shape=in_shape)
+        hsv_out = gen_image_ops.rgb_to_hsv(rgb_input_tensor)
+        self.assertEqual(out_shape, list(hsv_out.get_shape()))
+
+        hsv_out = self.evaluate(hsv_out)
+      self.assertEqual(out_shape, list(hsv_out.shape))
+
+  def testRGBToHSVGrad(self):
+    in_shape = [2, 20, 30, 3]
+    def f(x):
+      return gen_image_ops.rgb_to_hsv(x)
+
+    x = np.random.rand(2, 20, 30, 3).astype(np.float32)
```

Comment:
Initializing `x` with random values is likely to cause problems. If any of the (2*20*30) RGB values in the tensor happen to fall near a discontinuity, the test will fail. Better to use a constant value for `x` that still exercises the gradient function. There are a few ways to do this. You could manually pick a set of RGB values that exercise all the parts of your gradient equation; and include those values as a constant. Or you could add some code to detect and replace problematic RGB values.

Reply:
Resolved in 032bdc6.